mon-pix/app/routes/assessments/challenge.js

```diff
@@ -7,22 +7,35 @@ export default class ChallengeRoute extends Route {
   @service currentUser;
 
   async model(params) {
-    const store = this.store;
-
     const assessment = await this.modelFor('assessments');
     await assessment.certificationCourse;
-    const challengeId = params.challenge_id;
+    await assessment.answers;
+
+    let challenge;
+
+    const isBackToPreviousChallenge = parseInt(params.challenge_number) < assessment.answers.length;
+    if (isBackToPreviousChallenge) {
+      const challengeId = assessment.answers.toArray()[params.challenge_number].challenge.get('id');
```

Comment:
Zut, devancé.
J'allais proposer une petite modification du style : 
```suggestion
      const answers = assessment.answers.toArray()
      const challengeId = answers[params.challenge_number].challenge.get('id');
```